### ISSUE TITLE
Bootstrap the project

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/vshard#1]
Let's implement a proof-of-concept of sharding based on Virtual Buckets concept.

- Create vshard.router and vshard.storage module structure
- Use router.lua by @GeorgyKirichenko as a start point for vshard.router
- Implement replicaset configuration from https://github.com/tarantool/shard/issues/38:
```
local router = require('vshard.router')
router.cfg { --will be moved to box.cfg() sometime
    sharding = {
           { uri = '10.0.0.21:3301', replicaset=1, master = true },
           { uri = '10.0.0.22:3301', replicaset=1 },

           { uri = '10.0.0.21:3302', replicaset=2, master = true },
           { uri = '10.0.0.22:3302', replicaset=2 },

           { uri = '10.0.0.21:3303', replicaset=3, master = true },
           { uri = '10.0.0.22:3303', replicaset=3 },
    },
}
```
- router: parse configuration into Lua hash  { replicaset => [master, slave, ...] }, where master-slave are net.box objects. Use https://github.com/tarantool/shard/tree/dev for this code.
- storage: create table _vbucket to store bucked_id (the single field)
- router: on shard.cfg() connect to all servers and check for _vbucket table
- router: create routing table based on _vbucket contents from replicasets
- router: add a method shard.bootstrap() which will make initial redistribution of buckets over shards. The number of vbuckets can be hardcoded for now.
- storage: add a dispatcher function which will check bucket_id for every request and execute a Lua function with provided arguments

```
vshard.storage.dispatch = function(bucket_id, func, {arg1, arg2, ...}) end
```

- router: implement a function to call procedures by bucket_id:
  shard.call(key, func, {args}) => result
- Add tests
- Add CI

**Comment by rtsisyk** [comment]
>
>  *  Mean number of ranges in the cluster: CockroachDB calculates the mean number of ranges across the cluster and considers it the ideal range count for each node. For example, in a five-node cluster, with four nodes containing 20 ranges and the fifth node containing 25 ranges, the mean and ideal count per node would be 21.
>  *  Actual number of ranges on each node: If the number of ranges on a node is above the mean, the node moves enough ranges to other nodes to bring its range count closer to the mean. For example, continuing the scenario above, if 21 were the mean and ideal range count per node, the node with 25 ranges would move 1 range to each of the four other nodes, bringing each node to 21 ranges.

> Preventing Thrashing
> 
> Another issue we encountered early on was what we called “thrashing”. Our original heuristic of moving replicas around based just on the mean caused some ranges to keep moving back and forth, ad nauseum. This meant that the cluster would never reach equilibrium such that a lot of replicas were continuously being moved back and forth between nodes. This continuous unneeded movement uses up resources and can add latency to the system. To combat this, a buffer was added to both the high and low sides of the mean (5%). So only once a store was above the mean plus this threshold, was it considered for rebalancing and we only consider a rebalancing if moving a replica actually pushed the cluster as a whole closer to the mean.

It seems that our algorithm is simpler and move efficient than CockroachDB has.
